enhancements/console/samples.md

```diff
@@ -0,0 +1,415 @@
+2. A way to hide installed samples. Operators creating a sample resource should have a way to opt out of their samples.
+
+## Proposal
+
+### Workflow Description
+
+The Console Operator will provide a cluster-wide `ConsoleSample` CRD.
+
+Other operators like the OpenShift Serverless operator will install their own `ConsoleSample` CRs.
+
+The web console will automatically load all `ConsoleSample` CRs and show them in the sample catalog.
+
+Users can select one of samples and follow the import flows to create a demo application.
+
+### API Extensions
+
+The console operator automatically installs a new `ConsoleSample` CRD.
+
+### Risks and Mitigations
+
+#### The OpenShift Console is optional and could be disabled
+
+Other operators should expect that the `ConsoleSample` CRD is unavailable like the other `Console*` CRDs.
+
+### Drawbacks
+
+Another extra (but specific) custom resource just for console samples.
+
+## Design Details
+
+An "Import from Git" example of how we have it today as part of the Builder Images annotation:
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsoleSample
+metadata:
+  name: java-sample
+spec:
+  title: Java
+  abstract: Build and run Java applications using Maven and OpenJDK.
+  description: |
+    # About this sample
+    Build and run Java applications using Maven and OpenJDK.
+    # Links
+    * [Sample repository](https://github.com/jboss-openshift/openshift-quickstarts)
+  icon: data:image;base64,base64 encoded image
+  type: Source to Image
+  provider: Red Hat
+  tags:
+  - java
+  - jboss
+  - openjdk
+  source:
+    type: GitImport
+    gitImport:
+      url: https://github.com/jboss-openshift/openshift-quickstarts
+```
+
+An "Import container image" example:
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsoleSample
+metadata:
+  name: minimal-ubi-container
+spec:
+  title: Minimal UBI container
+  abstract: Test the minimal Red Hat Universal Base Image (UBI).
+  description: |
+    # About this sample
+    The Red Hat Universal Base Image is free to deploy on Red Hat or non-Red Hat platforms
+    and freely redistributable...
+  icon: data:image;base64,base64 encoded image
+  type: UBI Container
+  provider: Red Hat
+  source:
+    type: ContainerImport
+    containerImport:
+      image: registry.access.redhat.com/ubi8/ubi-minimal:8.8-860
+```
+
+### Attributes
+
+The following constraints are checked in the [openshift/api change](https://github.com/openshift/api/pull/1503) as well:
+
+1. `title` is the display name of the sample. It is required and can have a maximum length of **50** characters.
+2. `abstract` is a short teaser the sample. It is required and can have a maximum length of **100** characters.
+
+   Currently, the abstract is shown on the sample card tile below the title and provider
+3. `description` is required, has no maximum length, and is a README.md-like content for
```

Comment:
I think we should put a maximum length on this, say 2048 characters. Else you have to deal with an unbounded amount of text to be rendered in the console. 2048 characters should be enough right?

Reply:
The abstract will be shown on the card and is limited to 100 chars.

The description is like a README, it can also have quickly 3k or 4k characters. The console can use CSS `max-height: 1000px` to show a scrollbar if needed. Of course, I can add a 4k, 5k or 10k limit if you want, but I personally don't think that this is required.

Reply:
max-length 4096?

Reply:
Added the 4k max-length:

3. `description` is required and can have a maximum length of **4096** characters.
   It is a README.md-like content for additional information, links, pre-conditions, and other instructions.
   It will be rendered as **Markdown** so that it can contain line breaks, links, and other simple formatting.